play/src/containers/GameInProgressContainer.tsx

```diff
@@ -1,118 +1,60 @@
-  gameSession,
-  teamId,
-  currentState,
-  teamAvatar,
-  addTeamAnswerToTeamMember,
-  updateTeamScore,
-}: GameInProgressContainerProps) {
-  const [isPregameCountdown, setIsPregameCountdown] = useState<boolean>(true);
-  const currentQuestion =
-    gameSession?.questions[gameSession?.currentQuestionIndex ?? 0];
-  const currentTeam = gameSession?.teams?.find((team) => team.id === teamId);
-  // locally held score value for duration of gameSession, updates backend during each PHASE_X_RESULTS
-  const [score, setScore] = useState(currentTeam?.score ?? 0);
-  const leader = true;
-  const answerChoices = currentQuestion?.choices!.map((choice: IChoice) => ({ // eslint-disable-line @typescript-eslint/no-non-null-assertion
-    id: uuidv4(),
-    text: choice.text,
-    isCorrectAnswer: choice.isAnswer,
-    reason: choice.reason ?? '',
-  }));
-
-  const handlePregameTimerFinished = () => {
-    setIsPregameCountdown(false);
-  };
-
-  const handleUpdateScore = (inputScore: number) => {
-    updateTeamScore(teamId, inputScore);
-    setScore(inputScore);
+export default function GameInProgressContainer(props:GameInProgressContainerProps) {
+  const { apiClient } = props;
+  const { t } = useTranslation();
+  const [retry, setRetry] = React.useState<number>(0);
+  // text hoisted so that we can listen to changes in retry state and update according to returned value from custom hook
+  const [errorText, setErrorText] = React.useState<{title1:string, title2:string}>({title1: t('joingame.errormodal.title1'), title2: ''});
+  
+  // if user clicks retry on the error modal, increment retry state to force a rerender and another call to the api
+  const handleRetry = () => {
+    setErrorText({title1: 'Trying to reconnect...', title2: ''});
+    setRetry(retry + 1);
   };
+  
+  // loads game data from local storage
+  // if no game data, redirects to splashscreen
+  const pregameModel = useFetchLocalData();
+  // uses local game data to subscribe to gameSession 
+  // fetches gameSession first, then subscribes to data, finally returns object with loading, error and gamesession 
+  const  subscription = useFetchAndSubscribeGameSession(pregameModel.gameSessionId, apiClient, retry);
 
-  switch (currentState) {
-    case GameSessionState.CHOOSE_CORRECT_ANSWER:
-      return isPregameCountdown ? (
-        <PregameCountdown
-          handlePregameTimerFinished={handlePregameTimerFinished}
-        />
-      ) : (
-        <GameInProgress
-          {...gameSession}
-          teamAvatar={teamAvatar}
-          answerChoices={answerChoices}
-          teamId={teamId}
-          score={score}
-          addTeamAnswerToTeamMember={addTeamAnswerToTeamMember}
-        />
-      );
-    case GameSessionState.CHOOSE_TRICKIEST_ANSWER:
-    case GameSessionState.PHASE_1_DISCUSS:
-    case GameSessionState.PHASE_2_DISCUSS:
+  // this listens for changes to the retry state and updates the error text accordingly
+  React.useEffect(() => {
```

Comment:
this feels janky to me. I'm using it to listen for updates on the retry/subscription retry so that when the user retries, I can update the modal text from 'An error has occurred' to 'Retrying' and then back to 'An error has occurred' if the custom hook fails again. I feel like I'm putting state logic where it shouldn't go but I can't find another place to put it (originally it was down in the modal but I was getting a bunch of stale data).

Reply:
I've updated the code to remove this `useEffect` through the moving of all the text handling to the `useFetchAndSubscribeGameSession` hook. I think this is a lot cleaner than having to listen for the changes again here and the behaviour of the modal is nicer (it doesn't render on the retry click and then appears again when the error is thrown). I've updated the video so you can see it in action there.